docs/architecture/adr-038-state-listening.md

```diff
@@ -0,0 +1,607 @@
+# ADR 038: KVStore state listening
+
+## Changelog
+
+- 11/23/2020: Initial draft
+
+## Status
+
+Proposed
+
+## Abstract
+
+This ADR defines a set of changes to enable listening to state changes of individual KVStores and exposing these data to consumers.
+
+## Context
+
+Currently, KVStore data can be remotely accessed through [Queries](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/messages-and-queries.md#queries)
+which proceed either through Tendermint and the ABCI, or through the gRPC server.
+In addition to these request/response queries, it would be beneficial to have a means of listening to state changes as they occur in real time.
```

Comment:
Yes, and it would be great to enable many remote clients to listen to these changes...

In [another sdk](https://github.com/iov-one/weave), we wrote events on every set and delete. We didn't encode state, but it did provide a great way to subscribe to eg. changes in your bank balance